ext/AvizExt/viz/clustering.jl

```diff
@@ -29,24 +32,162 @@ function ADRIA.viz.ts_cluster!(g::Union{GridLayout,GridPosition}, data::Abstract
+
+    ADRIA.viz.map!(g, rs, data_stats; opts=opts, axis_opts=axis_opts)
 
     return g
 end
 
+"""
+    _cluster_colors(clusters::Vector{Int64})
+
+Vector of cluster colors.
+
+# Arguments
+- `clusters` : Vector of numbers corresponding to clusters
 
+# Returns
+Vector{RGBA{Float32}}
+"""
+function _clusters_colors(clusters::Vector{Int64})::Vector{RGBA{Float32}}
+    # Number of non-zero clusters
+    n_clusters = length(unique(filter(cluster -> cluster != 0, clusters)))
+
+    # Vector of clusters colors for non-zero clusters
+    clusters_colors = categorical_colors(:seaborn_bright, n_clusters)
+
+    # Assign color "black" to cluster == 0
+    rgba_black = parse(RGBA{Float32}, "transparent")
+    return [cluster == 0 ? rgba_black : clusters_colors[cluster] for cluster in clusters]
+end
+
+"""
+    _cluster_color(unique_cluster_colors::Vector{RGBA{FLoat32}}, clusters::Vector{Int64}, cluster::Int64)
+
+Color parameter for current cluster weighted by number of scenarios
+
+# Arguments
+- `unique_cluster_colors` : Vector of all cluster options that are being used
+- `clusters` : Vector of numbers corresponding to clusters
+- `cluster` : Current cluster
+
+# Returns
+Tuple{RGBA{Float32}, Float64}
+"""
+function _cluster_color(unique_cluster_colors::Vector{RGBA{Float32}},
+    clusters::Vector{Int64}, cluster::Int64)::Tuple{RGBA{Float32},Float64}
+    # Number of scenarios for a cluster
+    n_scens = count(clusters .== cluster)
+
+    # Compute line weight
+    color_weight = max(min((1.0 / (n_scens * 0.05)), 0.6), 0.1)
+
+    return (unique_cluster_colors[cluster], color_weight)
+end
+
+"""
+    _cluster_legend_params(clusters::Vector{Int64}, clusters_colors::Vector{RGBA{Float32}}, data_statistics::Vector{Float32})
+
+Color parameter for current cluster weighted by number of scenarios
+
+# Arguments
+- `clusters` : Vector of numbers corresponding to clusters
+- `cluster_colors` : Vector of all cluster options that are being used
+- `data_statistics` : Vector of statistics for each site (default is mean)
+
+# Returns
+Tuple{RGBA{Float32}, Float64}
+"""
+function _cluster_legend_params(clusters::Vector{Int64}, 
+    clusters_colors::Vector{RGBA{Float32}}, data_statistics::Vector{Float32})::Tuple
+    # Filter non-zero clusters from clusters, colors and data
+    non_zero_clusters = clusters .!= 0
+    clusters_filtered = clusters[non_zero_clusters]
+    colors_filtered = clusters_colors[non_zero_clusters]
+    statistics_filtered = data_statistics[non_zero_clusters]
+
+    # Fill legend entries colors
+    legend_entries = [PolyElement(color = color, strokecolor = :transparent) for 
+        color in unique(colors_filtered)]
+
+    # Fill legend labels
+    legend_labels = String[]
+    clusters_numbers = unique(clusters_filtered)
+    for cluster in clusters_numbers
+        stat_mean = mean(statistics_filtered[cluster .== clusters_filtered])
+        stat_mean_formatted = @sprintf "%.1e" stat_mean

```

Comment:
As mentioned, I think using formatted values (10, 100, 1K, 10K, 100K, 1M, etc) is easier for comparison than scientific notation

Reply:
This is a small comment though - if it seems too time consuming to get working, please ignore for now.